### ISSUE TITLE
Make placeholder optional

### ISSUE THREAD

**Issue by tjkirch** [joereynolds/vim-minisnip#44]
Some of my snippets are just plain text that I want to be able to insert more quickly.  No placeholders.  Currently I need to add "{{++}}" at the end of the snippet text or I get the error below.  It seems like the plugin could default to putting the cursor after the text if there are no placeholders.

```
Error detected while processing function minisnip#Minisnip[45]..<SNR>106_SelectPlaceholder:
line   12:
E486: Pattern not found: \V{{+\.\{-}+}}
Error detected while processing function minisnip#Minisnip:
line   45:
E171: Missing :endif
```

**Comment by arp242** [comment]
I use several snippets without placeholders, and have never encountered this error :thinking: Looking at the code for `s:SelectPlaceholder()`, this situation should be handled (that's what the `catch /E486:/` is for).

Can you confirm you're using the latest version, and share the snippet you're using, as well as your Vim version (first *two* lines from `:version`)?

**Comment by tjkirch** [comment]
> your Vim version

I just updated to latest and confirmed it made no difference:

```
VIM - Vi IMproved 8.1 (2018 May 18, compiled Jul 29 2019 09:46:30)
Included patches: 1-1773
```

> confirm you're using the latest version

Yep; tried to update, and no change there, still on b5bcaff0ddea1 from March 29, 2018.

> share the snippet you're using

I've tried a few; this is a simple reproducer:

```
$ cd ~/.vim/minisnip
$ cat _text_qwer
testing with no marker
```

I found something interesting while reproducing, though - there's a timing element involved.  If I have the blank `{{++}}` placeholder, a snippet always works with no errors.

If I *don't* have the placeholder, then I can cause the error reliably by typing the Tab that expands the snippet very quickly after typing the last letter of the marker text.  So, with the example above, if I have "qwe" on a line, then type "r" and "Tab" very quickly, I get the error every time.  I'm not sure how to time that precisely, but something like a 0.1-0.2 second delay makes it work reliably, but if I'm typing at a normal quick speed it can fail.

That struck me as odd, like it's probably due to my setup and something happening asynchronously in insert mode.  So, I bisected my settings and found that it was the "ale_completion_enabled" setting for https://github.com/w0rp/ale.  It doesn't seem to have a setting to enable completion only for certain languages or certain circumstances, but luckily for me, I stopped using it so I can just turn off that setting.

In any case, almost surely not a vim-minisnip issue!  Closing, sorry for the trouble.
